lib/pages/report.dart

```diff
@@ -0,0 +1,229 @@
+import 'package:capstone_app/repositories/reportRepository.dart';
+import 'package:capstone_app/style/appTheme.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_calendar_carousel/flutter_calendar_carousel.dart';
+import 'package:capstone_app/models/garage.dart';
+import 'package:capstone_app/repositories/garageRepository.dart';
+import 'package:capstone_app/repositories/accountRepository.dart';
+
+import 'package:capstone_app/services/auth.dart';
+
+class ReportPage extends StatefulWidget {
+  @override
+  ReportPageState createState() => ReportPageState();
+}
+
+class ReportPageState extends State<ReportPage> {
+  GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
+
+  ReportRepository _reportRepo = ReportRepository.getInstance();
+
+  AccountRepository _accountRepo = AccountRepository.getInstance();
+
+  GarageRepository _garageRepo = GarageRepository.getInstance();
+  List<Garage> _garages = List();
+  Future<List<Garage>> _garageFuture;
+
+  DateTime _selectedDate;
+  Garage currentSelectedValue;
+
+  TimeOfDay _time = TimeOfDay.now();
+  TimeOfDay picked;
+
+  void _reportGarage() {
+//    TimeOfDay newStartTime = _getTimeOfDayForIndex(_selectedTime.truncate());
+    DateTime reportTime = DateTime(_selectedDate.year, _selectedDate.month, _selectedDate.day, _time.hour, _time.minute);
+   if(currentSelectedValue != null){
+     _accountRepo.reportTicketForGarage(reportTime, currentSelectedValue);
+     print("Reported");
+   }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _selectedDate = _reportRepo.intervalStart;
+    _garageFuture = _garageRepo.getGarages().timeout(Duration(seconds: 30));
+
+    _garageFuture.then((List<Garage> garages) async {
+      setState(() {
+        _garages = garages;
+      });
+    });
+  }
+
+  Widget _typeFieldWidget(){
+    return Padding(
+      padding: EdgeInsets.only(
+          top: 5.0,
+          bottom: 5.0
+      ),
+
+              child: DropdownButton<Garage>(
+                hint: Text("Select Garage"),
+                value: currentSelectedValue,
+                onChanged: (Garage newValue) {
+                  setState(() {
+                    currentSelectedValue = newValue;
+                  });
+                  print(currentSelectedValue.name);
+                },
+                items: _garages.map((Garage value) {
+                  return DropdownMenuItem<Garage>(
+                    value: value,
+                    child: Text(value.name),
+                  );
+                }).toList(),
+              ),
+            );
+  }
+
+  Future<Null>selectTime(BuildContext context)async{
+    picked=await showTimePicker(

```

Comment:
If the cancel button on the time picker is clicked, _time gets set to null and trying to open the dialog again results in an unhandled exception